server/src/Lib.hs

```diff
@@ -0,0 +1,187 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Lib (app) where
+
+import Control.Monad
+import Control.Monad.IO.Class
+import Control.Monad.Trans.Except (ExceptT (..), runExceptT, withExceptT)
+import Data.ByteString (ByteString)
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy.Char8 as BSL
+import Data.Coerce (coerce)
+import Data.Either (isLeft)
+import Data.Maybe (fromMaybe)
+import Database.SQLite.Simple
+import Database.SQLite.Simple.FromField (FromField (..))
+import Database.SQLite.Simple.ToField (ToField (..))
+import GHC.IO.Exception (ExitCode (ExitSuccess))
+import Network.Wai.Handler.Warp (run)
+import Servant
+import qualified System.Directory as Dir
+import System.FilePath ((</>))
+import qualified System.Process as Proc
+
+-- | Top-level API. Examples:
+--   localhost:8081/ps
+--   localhost:8081/procps/ps?sys=aarch64-linux
+--  TODO API documentation
+type API =
+  Get '[JSON] NoContent
+    :<|> Capture "binary-name" BinaryName :> QueryParam "sys" Platform :> Get '[OctetStream] ByteString
+    :<|> Capture "package-name" PackageName :> Capture "binary-name" BinaryName :> QueryParam "sys" Platform :> Get '[OctetStream] ByteString
+
+newtype BinaryName = BinaryName {unBinaryName :: String}
+  deriving newtype (ToField, FromHttpApiData)
+
+newtype PackageName = PackageName {unPackageName :: String}
+  deriving newtype (Eq, Ord, FromField, FromHttpApiData, ToField)
+
+data Platform
+  = X86_64_Linux
+  | AARCH_64_Linux
+
+instance Show Platform where
+  show X86_64_Linux = "x86_64-linux"
+  show AARCH_64_Linux = "aarch64-linux"
+
+instance FromHttpApiData Platform where
+  parseQueryParam "aarch64" = pure AARCH_64_Linux
```

Comment:
```suggestion
  parseQueryParam "aarch64-linux" = pure AARCH_64_Linux
```

Reply:
This became 3d83a0df8e646268b8557d8b35d498d9f1a31b5c